package.json

```
@@ -4,18 +4,18 @@
   "description": "Manage large Open API specifications and automatically generate TypeScript definitions from them.",
   "main": "src/index.ts",
   "engines": {
-    "node": ">18.12.1"
+    "node": ">=18.12.1"
   },
   "scripts": {
     "test": "jest",
-    "dev": "node-dev src/index.ts",
-    "build": "tsc",
+    "dev": "node-dev src/write-bin-template.ts",
+    "build": "tsc --project tsconfig.build.json && node bin/write-bin-template.js",
     "format": "prettier -w .",
     "lint": "eslint --ignore-path .prettierignore \"**/*.ts\"",
     "lint:fix": "eslint --fix --ignore-path .prettierignore \"**/*.ts\"",
     "lint:check-rule-overlap": "npx eslint-config-prettier src/index.ts",
     "typecheck": "tsc --noEmit",
-    "postinstall": "npm run build",
+    "validate-templates": "npx swagger-cli validate ./bin/templates/json/[entryPoint].json & npx swagger-cli validate ./bin/templates/yaml/[entryPoint].yaml",
```

With a single ampersand, the process won't exit correctly if it fails.
```suggestion
    "validate-templates": "npx swagger-cli validate ./bin/templates/json/[entryPoint].json && npx swagger-cli validate ./bin/templates/yaml/[entryPoint].yaml",
```

---

🤔 hmm that's interesting. Is that in the case that the first one errors but not the second? I thought it would be ok to run these parallel

---

To be honest, I can't remember, I just deliberately sabotaged the src, so I guess the scenario was _both_ failing.

---

I've been reading into this. It may well be the way that the separate subshell terminates doesn't show the overall status as existing with a failure.

This could probably be worked around.

What can't be fixed is that this is one of the places where Windows doesn't play nice (which is probably why I haven't used a single ampersand in a shell command for a while)

Probably easiest to use something like `npm-run-all` instead.

---

Now fixed with the use of [concurrently](https://www.npmjs.com/package/concurrently), which IMHO had a nicer feature set and better documentation than [npm-run-all](https://www.npmjs.com/package/npm-run-all).